src/chaat/handler/handler.clj

```diff
@@ -30,17 +40,25 @@
   [db request]
   (let [params (:params request)
         {:keys [username password]} params
-        result (handler.validation/validate-signup-details username password)
+        result (handler.validation/validate-credentials username password)
         result (do-or-error result model.user/create db username password)]
```

Comment:
Can you not use the `until-err->` macro for this? Shadowing bindings like this is a smell imo, shouldn't be needed when you can use a threading macro instead

Reply:
We discussed my reasons for doing this at the office, and you suggested some good alternatives. I won't do this refactor now, but I am aware of the alternative options including the pedestal library. 

Next time we speak, I would like to spend a few more minutes discussing the details of the other option where we d not return detailed error messages for the failure case, and just return `nil` or let exceptions be thrown which ring will handle. 

I also want to checkout [failjure](https://github.com/adambard/failjure).